File: BACKEND/Shared/Dtos/DriverDto.cs
using System.ComponentModel.DataAnnotations;


namespace Shared.Dtos
{
    public class DriverRequest
    {
    
        [Required]
        [StringLength(16)]
        public required string LicenceNumber { get; set; }
        [Required]
        public EmployeeRequest? Employee { get; set; }
        [Required]
        public string PermisType { get; set; }
    }

    public class DriverResponse
    {
        public int Id { get; set; }
        public required string LicenceNumber { get; set; }
        public EmployeeResponse? Employee { get; set; }
        public string PermisType { get; set; }
    }
}

File: FRONTEND/src/app/shared/types/Dtos/driver.dto.ts
import { permisType } from "../../../features/Drivers/types/driver.model";
import { EmployeeRequest } from "./Employee.dto";

export interface DriverRequest{
    employee:EmployeeRequest;
    licenceNumber:string;
    permisType:permisType;
}


export interface DriverResponse{
    id:number;
    employee:EmployeeRequest;
    licenceNumber:string;
    permisType:permisType;
}